internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -142,7 +142,7 @@ func (s *Server) Register(obj *grpc.Server) {
 
 	// wrap resources with access filter
 	resourceState := s.Controller.Runtime().State().V1Alpha2().Resources()
-	resourceState = state.WrapCore(state.Filter(resourceState, resources.AccessPolicy(resourceState)))
+	resourceState = state.WrapCore(state.Filter(resourceState, resources.AccessPolicy(resourceState, false)))
```

Comment:
this works, but feels a bit like twisted flow.

as a proposal to simplify and streamline things, should we add a middleware which checks whether the peer is SideroLink, and injects `role.Admin` into the context?

this way in resource service things will "just work", while in methods which are only available over SideroLink we can assert on the role.Admin, and return the error otherwise?

Reply:
changed it, pls have another look.

Reply:
changed it to be that way.